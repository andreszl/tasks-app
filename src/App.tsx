import React, { useState, useEffect } from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { connect } from 'react-redux';
import { isEmpty } from 'lodash';

import routes from './routes';
import Layout from './components/organisms/layout/layout.organism';
import Login from './components/molecules/login/login.molecule';
import * as interfaces from './interfaces';
import actions from './actions';

function mapStateToProps(state: interfaces.redux.Store) {
	return {
		user: state.user,
	};
}

const { signIn } = actions.auth;

const mapDispatchToProps = {
	signIn,
};

type MapStateToProps = ReturnType<typeof mapStateToProps>;
type MapDispatchToProps = typeof mapDispatchToProps;

export interface Props extends MapStateToProps, MapDispatchToProps {
}

const App = ({ user, signIn }: any) => {
	const [logged, setLogged] = useState<boolean>(false);
	useEffect(() => {
		signIn().then((result: any) => {
			if (result) {
				setLogged(true);
			}
		});
	}, []);

	useEffect(() => {
		if (isEmpty(user._id)) {
			setLogged(false);
		}
	}, [user._id]);

	const onSignIn = (status: boolean) => {
		setLogged(status);
		return status;
	};

	return (
		<div className="bg-blue-light z-index-1001">
			<Router>
				<div>
					<Routes>
						{
							routes.map(({ path, component, useLayout, layout, props, requireLogin }, index) => {
								const key = `${path}-${index}`;
								const RouteComponent = component;

								if (!RouteComponent) return null;

								let CustomLayout = Layout;

								if (useLayout && layout) {
									CustomLayout = layout;
								}

								const CustomComponent = () => {
									if (requireLogin) {
										useLayout = false;
									}
									return (
										<div>
											{requireLogin && !logged ? (
												<div className="vh-100">
													<Login onSignIn={onSignIn} />
												</div>
											) : <RouteComponent {...props} />}
										</div>
									);
								};

								return (
									<Route
										key={key}
										path={path}
										element={(useLayout && logged ? (
											<CustomLayout>
												<CustomComponent />
											</CustomLayout>
										) : <CustomComponent />)}
									/>
								);
							})
						}
					</Routes>
				</div>
			</Router>
		</div>
	);
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
