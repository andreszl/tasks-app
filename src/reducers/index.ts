import { combineReducers } from 'redux';
import user from './auth/auth.reducer';

const reducers = combineReducers({
	user,
});

export default reducers;
