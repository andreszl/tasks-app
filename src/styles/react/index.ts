import * as drawer from './drawer.styles';

export {
	drawer,
};
