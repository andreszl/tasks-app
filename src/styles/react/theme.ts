import { createTheme, responsiveFontSizes } from '@mui/material/styles';

const theme = createTheme({
	palette: {
		primary: {
			main: '#4071B8',
		},
		secondary: {
			main: '#FFFFFF',
		},
		warning: {
			main: '#db7878',
		},
	},
	typography: {
		fontFamily: [
			'Montserrat',
			'Roboto',
			'"Helvetica Neue"',
			'Arial',
			'sans-serif',
		].join(','),
	},
});

export default responsiveFontSizes(theme);
