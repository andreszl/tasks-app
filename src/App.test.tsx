import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducers from './reducers';
import App from './App';

const store = createStore(
	reducers,
	applyMiddleware(thunk),
);

describe('App Component', () => {
	const TestComponent = (
		<Provider store={store}>
			<App />
		</Provider>
	);

	it('should render it', () => {
		const { container } = render(TestComponent);
		expect(container).toMatchSnapshot();
	});
});
