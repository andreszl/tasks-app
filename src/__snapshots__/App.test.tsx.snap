// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render it 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="bg-blue-light z-index-1001"
  >
    <div>
      <div>
        <div
          class="vh-100"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`App Component test 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="bg-blue-light z-index-1001"
  >
    <div>
      <div>
        <div
          class="vh-100"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`MyComponent deber√≠a cambiar el estado de logged correctamente cuando se llama a onSignIn 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="bg-blue-light z-index-1001"
  >
    <div>
      <div>
        <div
          class="vh-100"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`MyComponent renders correctly 1`] = `ShallowWrapper {}`;

exports[`renders learn react link 1`] = `
<div
  aria-hidden="true"
>
  <div
    class="bg-blue-light z-index-1001"
  >
    <div>
      <div>
        <div
          class="vh-100"
        />
      </div>
    </div>
  </div>
</div>
`;
